prow/cmd/deck/jobs.go

```diff
@@ -157,64 +133,26 @@ func (ja *JobAgent) tryUpdate() {
 	}
 }
 
-type byStartTime []Job
-
-func (a byStartTime) Len() int           { return len(a) }
-func (a byStartTime) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
-func (a byStartTime) Less(i, j int) bool { return a[i].st.After(a[j].st) }
-
 func (ja *JobAgent) update() error {
 	pjs, err := ja.kc.ListProwJobs(nil)
 	if err != nil {
 		return err
 	}
-	var njs []Job
-	njsMap := make(map[string]Job)
+	var njs []pjutil.Job
+	njsMap := make(map[string]pjutil.Job)
 	njsIDMap := make(map[string]map[string]kube.ProwJob)
 	for _, j := range pjs {
-		buildID := j.Status.BuildID
-		nj := Job{
-			Type:    string(j.Spec.Type),
-			Repo:    fmt.Sprintf("%s/%s", j.Spec.Refs.Org, j.Spec.Refs.Repo),
-			Refs:    j.Spec.Refs.String(),
-			BaseRef: j.Spec.Refs.BaseRef,
-			BaseSHA: j.Spec.Refs.BaseSHA,
-			Job:     j.Spec.Job,
-			Context: j.Spec.Context,
-			Agent:   j.Spec.Agent,
-			ProwJob: j.Metadata.Name,
-			BuildID: buildID,
-
-			Started:     j.Status.StartTime.Format(time.Stamp),
-			State:       string(j.Status.State),
-			Description: j.Status.Description,
-			PodName:     j.Status.PodName,
-			URL:         j.Status.URL,
-
-			st: j.Status.StartTime,
-			ft: j.Status.CompletionTime,
-		}
-		if !nj.ft.IsZero() {
-			nj.Finished = nj.ft.Format(time.RFC3339Nano)
```

Comment:
submit-queue at least is parsing these (I think the only use is something hacky I added to find the most recent presubmits of various types), possibly other places. we should make sure they will also handle this correctly

Reply:
Updated. I can't find anything else that tries to parse time using this format.